projects/canopy/src/lib/filter-container/filter-container.component.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { TestBed, waitForAsync } from '@angular/core/testing';
+import {
+  DefaultRenderComponent,
+  MockComponents,
+  MockedComponentFixture,
+  MockRender,
+} from 'ng-mocks';
+import { DebugElement } from '@angular/core';
+
+import { LgButtonComponent, LgButtonToggleDirective } from '../button';
+import { keyName } from '../utils/keyboard-keys';
+
+import {
+  LgFilterContainerComponent,
+  LgFilterContainerPanelComponent,
+  lgFilterContainerPanelIdPrefix,
+  lgFilterContainerToggleIdPrefix,
+} from './';
+
+describe('LgFilterContainerComponent', () => {
+  let component: DefaultRenderComponent<LgFilterContainerComponent>;
+  let fixture: MockedComponentFixture<LgFilterContainerComponent>;
+  let debugElement: DebugElement;
+  let el: HTMLElement;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        LgFilterContainerComponent,
+        LgButtonToggleDirective,
+        MockComponents(LgFilterContainerPanelComponent, LgButtonComponent),
+      ],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = MockRender(`
+      <lg-filter-container>
+        <button id="test" lg-button variant="secondary-dark" lgButtonToggle>Filters</button>
+
+        <lg-filter-container-panel></lg-filter-container-panel>
+      </lg-filter-container>
+    `);
+
+    component = fixture.componentInstance;
+    debugElement = fixture.debugElement;
+    el = debugElement.children[0].nativeElement;
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have the default class', () => {
+    expect(el.getAttribute('class')).toContain('lg-filter-container');
+  });
+
+  it('should set the id and aria-controls of the toggle', () => {
+    expect(component.filterContainerToggle.id).toBe(
+      `${lgFilterContainerToggleIdPrefix}${component['uniqueId']}`,
+    );
+
+    expect(component.filterContainerToggle.ariaControls).toBe(
+      `${lgFilterContainerPanelIdPrefix}${component['uniqueId']}`,
+    );
```

Comment:
Just wondering if we need a test to ensure the ID is unique every time? 🤔 
E.g. perhaps creating two filter containers, and checking that the first one is `0`, and the second is `1`.

Reply:
Not sure to be honest, we have this logic in all the other components and we have never tested it - it's just a simple:

```
let nextUniqueId = 0;

...

export class ... {
  private uniqueId = nextUniqueId++;
```

Reply:
Yeah I did think that we haven't tested it before. Ideally, I think it would be good, as we test simple things like adding classes etc, so it makes sense that we'd check that the ID has been incremented to avoid duplicates, but it's definitely not a show stopper. Happy to approve anyway :)